### PR TITLE
4334: added dobi root script and matching bash script

### PR DESCRIPTION
**Pull request by FlorianReinhold**
first initial dobi based build structure.

### CODE REVIEW

File: dobi.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+set -o errexit   # abort on nonzero exitstatus
+set -o pipefail  # don't hide errors within pipes
+
+# set project name
+export PROJECT_NAME=bb-buildingblock
+
+# check argument count
+if [ -z ${1} ]; then
+    echo "no arguments!"
+    exit 1
```
Comment (mlilien) [inline dobi.sh:10]: i suggest using "list" argument instead

File: dobi.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+fi
+
+# extend list command
+if [ $1 = "list" ]; then
```
Comment (mlilien) [inline dobi.sh]: inconsistent usage of $1 vs ${1}

File: dobi.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    echo "Predifiend Resources:"
+    echo "  build   [--->] ${PROJECT_NAME}-build"
+    echo "  test    [--->] ${PROJECT_NAME}-test"
+    echo "  deploy  [--->] ${PROJECT_NAME}-deploy"
+    exec dobi list
+    exit 0
+fi
+
+# set dobi defaults
+if [ $1 = "test" ] || [ $1 = "build" ] || [ $1 = "deploy" ]; then
```
Comment (mlilien) [inline dobi.sh]: inconsistent usage of $1 vs ${1}

**Review by mlilien — CHANGES_REQUESTED** [review]

File: dobi.sh (hunk 3 above)
Comment (siredmar) [reply dobi.sh]: Right, as i learned always prefer ${var} over $var

File: dobi.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+# ===================================================
+
+meta:
+  project: bb-buildingblock
+
+# ===================================================
+# alias
+# ===================================================
+
+alias=bb-buildingblock-build:
```
Comment (siredmar) [inline dobi.yaml:5]: can those 'bb-buildingblock' be created form that project variable or even the env PROJECT_NAME?

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1,6 @@
+#general
+build
+gen
```
Comment (siredmar) [inline .gitignore:3]: what is 'gen' used for?

File: dobi.yaml [hunk 6]
```diff
@@ -0,0 +1,25 @@
+# ===================================================
+# meta infos
+# ===================================================
+
+meta:
+  project: bb-buildingblock
```
Comment (siredmar) [inline dobi.yaml]: meta section does not work for including subprojects. So if you intend this to be included as a subproject, this won't work.

File: dobi.yaml (hunk 4 above)
Comment (FlorianReinhold) [reply dobi.yaml:5]: no, there is no possibility to set config vars in aliases.

File: dobi.sh [hunk 7]
```diff
@@ -3,29 +3,12 @@
...
 set -o pipefail  # don't hide errors within pipes
 
-# set project name
-export PROJECT_NAME=bb-buildingblock
-
 # check argument count
 if [ -z ${1} ]; then
     echo "no arguments!"
+    exec dobi --filename meta.yaml list
     exit 1
```
Comment (mlilien) [inline dobi.sh:10]: is this part reachable? "exit 1" ?

**Review by mlilien — CHANGES_REQUESTED** [review]

**Review by siredmar — DISMISSED** [review]

File: dobi.sh [hunk 8]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+set -o errexit   # abort on nonzero exitstatus
```
Comment (siredmar) [inline dobi.sh:3]: test

File: meta.yaml [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+
+meta:
+  project: bb-buildingblock
+  include:
+  - dobi.yaml
+
+# ===================================================
+# alias
+# ===================================================
+
```
Comment (siredmar) [inline meta.yaml:13]: remote empty lines

File: meta.yaml [hunk 10]
```diff
@@ -0,0 +1,30 @@
+# ===================================================
+# meta infos
+# ===================================================
+
```
Comment (siredmar) [inline meta.yaml:4]: remote empty lines

File: dobi.yaml [hunk 11]
```diff
@@ -0,0 +1,18 @@
+# ===================================================
+# alias
+# ===================================================
+
```
Comment (siredmar) [inline dobi.yaml:4]: remote empty lines

**Review by siredmar — COMMENTED** [review]
Test

File: dobi.sh (hunk 8 above)
Comment (siredmar) [reply dobi.sh:3]: DO your homework

File: dobi.yaml (hunk 11 above)
Comment (siredmar) [inline dobi.yaml:4]: remove newlines

File: dobi.yaml [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+alias=bb-buildingblock-build:
+  tasks: []
+  annotations:
+    description: "-> build all content (not implemented yet)"
+
```
Comment (siredmar) [inline dobi.yaml:9]: remove newlines

File: dobi.sh [hunk 13]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
```
Comment (siredmar) [inline dobi.sh:2]: comment

File: dobi.sh [hunk 14]
```diff
@@ -0,0 +1,14 @@
...
+set -o errexit   # abort on nonzero exitstatus
+set -o pipefail  # don't hide errors within pipes
+
```
Comment (siredmar) [inline dobi.sh:5]: approve

File: dobi.sh [hunk 15]
```diff
@@ -0,0 +1,14 @@
...
+# check argument count
+if [ -z ${1} ]; then
+    echo "no arguments!"
```
Comment (siredmar) [inline dobi.sh:8]: request change

**Review by siredmar — CHANGES_REQUESTED** [review]
asdfasdfas

**Review by siredmar — COMMENTED** [review]
empty review

**Review by siredmar — APPROVED** [review]

File: dobi.sh (hunk 7 above)
Comment (FlorianReinhold) [reply dobi.sh:10]: you are right, thx! changed it.

**Review by mlilien — APPROVED** [review]

**Pull request merged**